### DISCUSSION TITLE
Ice 0.10.0 (Beta 6)

### DISCUSSION THREAD

**Discussion by jordanbaird** [jordanbaird/Ice#192] [Announcements]
## What's New
* You can now right click items in the Ice Bar
* Ice Bar now dynamically changes its color depending on its location
* Menu bar appearance can now be inset on screens with a notch
* Add ability to disable the always-hidden section
* Ice now uses the GPL-3.0 license

## Bug Fixes
* Fixed menu bar item image capture
* Items and their images are no longer cached while being moved
* Items are no longer arranged while the mouse is moving or dragging
* Fix application menus occasionally not hiding
* Prevent appearance overlays from moving to another screen when one screen has a fullscreen space

**Full Changelog**: https://github.com/jordanbaird/Ice/compare/0.10.0-beta.5...0.10.0-beta.6

<hr /><em>This discussion was created from the release <a href='https://github.com/jordanbaird/Ice/releases/tag/0.10.0-beta.6'>Ice 0.10.0 (Beta 6)</a>.</em>

**Comment by fleytman** [comment]
>Add ability to disable the always-hidden section

Ice bar stops working if you click on disable always-hidden section
Case for reproduce 

1. Open ice and make sure that the ice bar is working
2. Disable the always-hidden section
3. Restart ice
4. Call the ice bar


https://github.com/jordanbaird/Ice/assets/1923645/7900bc25-8ed8-48e5-9bfc-8bc44cf772d1

**Reply by jordanbaird** [reply]
I noticed this too after posting the release. Fixed in #193

**Comment by alexistrn** [comment]
Hello, 

Don't know if it's related, but this bug appeared just after beta install (I have the feeling that this appeared when I was moving or leaning the Macbook), continued even after multiple startups, and completely disappeared after deinstall of the beta. I haven't really done anything in between. I prefer to mention it, although I don't know how it could be related, other than via screen recording necessary for Ice to function ?

I don't touched anything during the screen recording, just tried to move trackpad. First time seeing this 😂 

https://github.com/jordanbaird/Ice/assets/112898349/3b2f8f45-efba-4b85-a6af-a8d5c710f801

**Reply by jordanbaird** [reply]
Odd 🤔. I can't imagine it would have anything to do with Ice. It doesn't install any additional tools that would be running in the background to cause this. Have you tried installing again to see if it still happens?

**Reply by alexistrn** [reply]
I have no idea either...🤔 honestly it doesn't matter, when I uninstalled the beta the problem was (apparently) resolved. I simply preferred to share it with you, just in case. The coincidence seemed a little too big to me ☺️ I will try a new install of the beta to see what happen.

**Reply by alexistrn** [reply]
It seems I was unable to reproduce the issue. However, I installed the beta by unzipping it in the "downloads" folder, and dragged the app to the application folder where the stable version was still located. Maybe that caused something, no idea.

**Comment by Xytronix** [comment]
Beta 5 and 6 are a huge step forward. Thank you! In full screen mode, Ice Bar should automatically hide or show the menu bar as long as Ice Bar is open. Additionally, I experienced a behavior that Ice Bar moves its position over the menu bar when it is open for a period of time. A timer to hide Ice Bar after a while could be a great QoL in this regard.

<img width="1540" alt="image" src="https://github.com/jordanbaird/Ice/assets/32957125/1189f522-cb99-4020-a279-2c63f780a342"> <img width="1701" alt="image" src="https://github.com/jordanbaird/Ice/assets/32957125/62fb3727-ddd8-4e25-bfe6-44e3535dae20"> 

Would you add Ice Bar as a separate section that can mirror the hidden and always-hidden parts? It's much more convenient for everyone to use Ice Bar because it mirrors the hidden part, although in some situations it's better to have separate parts. 

Do you plan to add menu bar groups and focus mode integration in an upcoming update?

**Reply by jordanbaird** [reply]
I'm not sure if there's a way to show and hide the menu bar without changing the system setting. I'll look into it. Does the behavior where the Ice Bar changes its position only happen in fullscreen mode, or is it all the time?

As for the second part, if I'm understanding correctly, I believe the Ice Bar can already do what you're asking. Either right-click the Ice icon and click "Show the Always-Hidden Section", or hold the Option key and click the Ice icon (or an empty space in the menu bar, if you have that setting enabled).

And yes, those features are both planned. The immediate next step after the Ice Bar is profiles. As that gets further along, I'll decide exactly where to go next. Probably triggers, which focus mode integration will be part of.

**Reply by Xytronix** [reply]
Yes, it happens only when I am being in full-screen mode.

For the second part, an additional layout under Menu Bar Items:
![image](https://github.com/jordanbaird/Ice/assets/32957125/b28f80bf-d118-4364-9140-3a0d5ee56078)
If users do want to set up the Ice Bar individually from the rest of the layouts.

**Reply by jordanbaird** [reply]
That's an interesting idea. Do you want to create a feature request issue for it?

**Reply by Xytronix** [reply]
done https://github.com/jordanbaird/Ice/issues/208

**Comment by alexistrn** [comment]
Hello ! I installed Ice directly from the "Applications" directory and when I want to update it, I get this error message. Is this the expected behavior, so do we necessarily need to unzip the file from the download folder first ? When Ice is unzipped from the "download" directory, I do get a message indicating that I am on a beta version which is currently the latest release.

<img width="283" alt="Capture d’écran 2024-07-05 à 18 48 59" src="https://github.com/jordanbaird/Ice/assets/112898349/5ea50ac4-dd44-4eba-8c37-8552c09d5032">

**Reply by jordanbaird** [reply]
This is a problem with macOS putting a "quarantine" on files that are added to the Applications folder without being dragged from Finder. If you use Terminal or some other program to move the file, or if you unzip the contents directly into the Applications folder, macOS will do this.

**Comment by 0rly0wl** [comment]
Works great and ready for release

**Comment by EvansMatthew97** [comment]
Using version 0.10.0-beta.6. When playing full screen games with custom cursors like League of Legends, the  cursor changes back to the default macOS cursor periodically. Exiting Ice while playing fixes the issue.
